goldmane/pkg/daemon/daemon.go

```diff
@@ -66,14 +66,23 @@ type Config struct {
 	PushIndex int `json:"push_index" envconfig:"PUSH_INDEX" default:"30"`
 }
 
-func Run() {
+func ConfigFromEnv() Config {
 	// Load configuration from environment variables.
 	var cfg Config
 	if err := envconfig.Process("", &cfg); err != nil {
 		logrus.WithError(err).Fatal("Failed to load configuration from environment")
 	}
 
 	utils.ConfigureLogging(cfg.LogLevel)
+
+	return cfg
+}
+
+func (cfg Config) ConfigureLogging() {
```

Comment:
Don't think we need this new function (and it's never being called)

Reply:
Sooo this is preemptive (and actually meant to replace the utils.ConfigureLogging up above.

If we're creating the config manually (as opposed to from the env), mostly for tests, then we want to ensure that we can configure logging.

Reply:
idk, feels like extra code for no benefit to me. We already have a ConfigureLogging that takes an argument... so why do we need a bound method that calls the utility function?

Ultimately it doesn't really matter, but I don't see the benefit. If we're creating the config manually we can just call the function that already exists?

Reply:
> We could but that means the caller needs to know the internal details of this. If we follow this pattern and we have 20 different test suites that create the Config struct then directly call utils.ConfigureLogging 

None of the test suites reference Config at all, because none of the test suites actually run the Daemon.

Reply:
Just removed it.